l5kit/l5kit/planning/utils.py

```diff
@@ -26,41 +54,19 @@ def _get_sides(bbox: Polygon) -> Tuple[LineString, LineString, LineString, LineS
     )
 
 
-def within_range(ego_centroid: np.ndarray, ego_extent: np.ndarray, agents: np.ndarray) -> np.ndarray:
-    agent_centroids = agents["centroid"]
-    agent_extents = agents["extent"]
-    distance = np.linalg.norm(ego_centroid - agent_centroids, axis=-1)
-    max_range = 0.5 * (np.linalg.norm(ego_extent[:2]) + np.linalg.norm(agent_extents[:, 2], axis=-1))
-    return agents[distance < max_range]
-
+def within_range(ego_centroid: np.ndarray, ego_extent: np.ndarray,
+                 agent_centroid: np.ndarray, agent_extent: np.ndarray) -> np.ndarray:
+    """This function will check if the agent is within range of the ego. It accepts
+    as input a vectorized form with shapes N,D or a flat vector as well with shapes just D.
 
-def detect_collision(
-        pred_centroid: np.ndarray, pred_yaw: float, pred_extent: np.ndarray, target_agents: np.ndarray
-) -> Tuple[str, str]:
-    """
-    Computes whether a collision occurred between ego and any another agent.
-    Also computes the type of collision: rear, front, or side.
-    For this, we compute the intersection of ego's four sides with a target
-    agent and measure the length of this intersection. A collision
-    is classified into a class, if the corresponding length is maximal,
-    i.e. a front collision exhibits the longest intersection with
-    egos front edge.
+    :param ego_centroid: the ego centroid (shape: N, 2)
+    :param ego_extent: the ego extent (shape: N, 3)
+    :param agent_centroid: the agent centroid (shape: M, 2)
```

Comment:
can this work when N!=M?  locally the line
```python
np.linalg.norm(ego_centroid - agent_centroid, axis=-1)
```
crashes when they are different.

Can you add a test for that in `test_within_range`?

Reply:
Don't know if understood it correctly, but this function should be the same as the original, do you want me to change the behavior of it ?